zebra/zebra_nb_config.c

```diff
@@ -1087,6 +1087,50 @@ int lib_interface_zebra_shutdown_destroy(struct nb_cb_destroy_args *args)
 	return NB_OK;
 }
 
+/*
+ * XPath: /frr-interface:lib/interface/frr-zebra:zebra/mpls
+ */
+int lib_interface_zebra_mpls_modify(struct nb_cb_modify_args *args)
+{
+	struct interface *ifp;
+	bool mpls;
+	struct zebra_if *if_data;
+
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	ifp = nb_running_get_entry(args->dnode, NULL, true);
+	if_data = ifp->info;
+	mpls = yang_dnode_get_bool(args->dnode, NULL);
+
+	if (mpls)
+		if_data->mpls_config = IF_ZEBRA_DATA_ON;
+	else
+		if_data->mpls_config = IF_ZEBRA_DATA_OFF;
+
+	dplane_intf_mpls_modify_state(ifp, mpls);
+
+	return NB_OK;
+}
+
+int lib_interface_zebra_mpls_destroy(struct nb_cb_destroy_args *args)
+{
+	struct interface *ifp;
+	struct zebra_if *if_data;
+
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	ifp = nb_running_get_entry(args->dnode, NULL, true);
+	if_data = ifp->info;
+
+	if_data->mpls_config = IF_ZEBRA_DATA_UNSPEC;
+
+	dplane_intf_mpls_modify_state(ifp, false);
```

Comment:
so ... this is sort of a problem we've had in this area. a 'destroy' should mean 'no configuration', but this looks like "configured to 'off'".

Reply:
you are right. I added a commit to fix this issue.